compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/handlers/IndexedGetIterationHandlers.kt

```diff
@@ -75,7 +78,20 @@ internal class ArrayIterationHandler(context: CommonBackendContext) : IndexedGet
     private val supportsUnsignedArrays = context.optimizeLoopsOverUnsignedArrays
 
     override fun matchIterable(expression: IrExpression) =
-        expression.type.run { isArray() || isPrimitiveArray() || (supportsUnsignedArrays && isUnsignedArray()) }
+        expression.type.run { isArray() || isPrimitiveArray() || (supportsUnsignedArrays && isUnsignedArray()) } || expression.run {
```

Comment:
Minor: I'd rather write it as 
```
expression.type.run { ... } ||
    expression.run { ... }
```
otherwise it's somewhat harder to understand that the last `||` is not actually a part of condition under `expression.type.run { ... }`.